### DISCUSSION TITLE
Multiple Lora

### DISCUSSION THREAD

**Discussion by DEAD-GOST** [sandeepmistry/arduino-LoRa#528] [Q&A]
Is this Compitable with 1 Receiver and multiple transmitters.
I mean will the message won't collide. or I have to implement the Network Protocol myself

**Comment by IoTThinks** [comment]
You need to implement the protocol on your own.

This library is just sending and receiving raw messages.

**Reply by DEAD-GOST** [reply]
what about that Gateway and Node example will that help ?

**Reply by IoTThinks** [reply]
No.
Collision will still happen with multiple modes.

**Reply by DEAD-GOST** [reply]
which protocol will be best

**Reply by IoTThinks** [reply]
Use this library if you have a few nodes.
Then add what you want.
Can try master to poll nodes to reduce collision.

Or go for LoRaWAN which is much more complicated.

**Reply by morganrallen** [reply]
@IoTThinks what are you talking about? This library will support multiple devices no problem.

@DEAD-GOST Because this library is very low level and only send raw packets it's up to you to implement your own addressing system. The `LoRaDuplexCallback` example shows this in a very basic way where the first `byte` of each packet is the `destination`, the next is the `source` and the rest is the data packet. The receiving side can then parse this packet to determine where it is destined.

**Reply by IoTThinks** [reply]
I never say this library can not support multiple devices.
Read again.

However, if not adding custom algorithm, no Listen before talk or collision detection or retransmission is available. Therefore, collision will happen and is not handled.

Unless either using polling, few devices or low sending rate.

**Reply by DEAD-GOST** [reply]
So, collision Avoidance protocol would be best right. As I can't detect collision.

**Reply by DEAD-GOST** [reply]
Also, what should be wait time. Since , Tt < 2Tp

**Reply by DEAD-GOST** [reply]
> @IoTThinks what are you talking about? This library will support multiple devices no problem.
> 
> @DEAD-GOST Because this library is very low level and only send raw packets it's up to you to implement your own addressing system. The `LoRaDuplexCallback` example shows this in a very basic way where the first `byte` of each packet is the `destination`, the next is the `source` and the rest is the data packet. The receiving side can then parse this packet to determine where it is destined.

So, how many devices can it handle

**Reply by IoTThinks** [reply]
> So, how many devices can it handle

How many nodes do you have?
And how frequent do you want to send the messages? E.g: 1 message / minute...

I love to start everything simple first especially at small scale.

**Comment by DEAD-GOST** [comment]
I am trying, to make Lora comunication for agriculture, 1 message per 3-5
min

**Reply by DEAD-GOST** [reply]
We can try, CDSM/CA (Collision Avoidance)

**Reply by IoTThinks** [reply]
> I am trying, to make Lora comunication for agriculture, 1 message per 3-5 min

For agriculture, the sensor value is not changing so fast. Even for every 30mins or 1 hour.
Go ahead with the default example. 

No need to make thing complicated.

**Comment by DEAD-GOST** [comment]
Actually, what if we are targeting a small-small section of farm, like filling up, or when it is over flow

**Reply by pubced** [reply]
You have various solutions:
Open loop mode (more simple)
- repeat each emission several times (4 times) a few seconds apart,
- or use an continuous measure/emission on upper frequency (each 10 minutes by ex) that necessary on reception,
- on receipt, drop all non-compliant message (via error check (CRC,...)) and keep only last good one for use whan needed (eah hour).
Of course, this solution increase your frequency payload, so use it with parsimony.

Or you can use feedback mode via hand-shake:
- the transmitter send one message,
- the receiver (only one if you have several) retransmit the message after recept (hand-shake),
- the transmitter receive this hand-shake: it's mean that the message has been correctly received,
- if the transmitter don't receive hand-shake before TIMEOUT, it retransmit again the initial message...

I use these two solutions and it works very well.

**Reply by Anonymouse-X** [reply]
Can you give example coding of the feedback mode via hand-shake? Thank you!
